org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaIndenter.scala

```diff
@@ -1518,10 +1517,9 @@ class ScalaIndenter(
    */
   private def isStringOrCharLiteralAssignment(referenceTokenPos: Int, offset: Int) = {
     val restOfTheLine = document.get(referenceTokenPos, Math.max(offset - referenceTokenPos, 0))
-    val charLit = """'(.|\\\w|\\\\)'"""
-    val stringLit = """"(\\"|[^"])*")"""
-
-    val regex = s"(${charLit}|${stringLit}$$".r
+    val charLit = "'.'"
+    val stringLit = "\".*\""
```

Comment:
About the regex:

```scala
// matches immediately
scala> "a"*51+"a" matches "(aa|aa)*"
res0: Boolean = true

// takes a long times to compute a result
scala> "a"*52+"a" matches "(aa|aa)*"
res1: Boolean = false

// Again, matches immediately
scala> "a"*52+"a" matches "(aa)*"
res2: Boolean = false
```
In this example "(aa|aa)*" is not optimized to "(aa)*" and because the Java Regex engine uses recursive backtracking which has [exponential running time](http://stackoverflow.com/questions/8887724/why-can-regular-expressions-have-an-exponential-running-time) and therefore do overflow at some point.

I think the previous regex overflowed, because the alternative of `\\"|[^"]` is recursively applied for every character (similar as in the above example). My new regex does not have `"\".*\""` doesn't have an alternative anymore and therefore should match sequentially.